flink-metrics/flink-metrics-prometheus/src/test/java/org/apache/flink/metrics/prometheus/PrometheusReporterTest.java

```diff
@@ -271,27 +228,16 @@ void registeringSameMetricTwiceDoesNotThrowException() {
 
     @Test
     void cannotStartTwoReportersOnSamePort() throws Exception {
-        ReporterSetup setup1 = createReporterSetup("test1", portRangeProvider.next());
-
-        int usedPort = ((PrometheusReporter) setup1.getReporter()).getPort();
-
-        try {
-            assertThatThrownBy(() -> createReporterSetup("test2", String.valueOf(usedPort)))
-                    .isInstanceOf(Exception.class);
-        } finally {
-            setup1.getReporter().close();
-        }
+        assertThatThrownBy(
+                        () ->
+                                new PrometheusReporter(
+                                        Collections.singleton(reporter.getPort()).iterator()))
+                .isInstanceOf(Exception.class);
     }
 
     @Test
     void canStartTwoReportersWhenUsingPortRange() throws Exception {
-        String portRange = portRangeProvider.next();
-
-        ReporterSetup setup1 = createReporterSetup("test1", portRange);
-        ReporterSetup setup2 = createReporterSetup("test2", portRange);
-
-        setup1.getReporter().close();
-        setup2.getReporter().close();
+        new PrometheusReporter(portRangeProvider.next()).close();
```

Comment:
I guess, we're testing different things here: The old implementation used the same range. The new test implementations runs on two different ranges (the iterator increases the base of the next range by 100)...